lib/grpc/GrpcServer.ts

```diff
@@ -70,6 +85,56 @@ class GrpcServer {
       });
     });
   }
+
+  private generateCertificate = (tlsCertPath: string, tlsKeyPath: string) => {
+    const keys = pki.rsa.generateKeyPair(1024);
+    const cert = pki.createCertificate();
+
+    cert.publicKey = keys.publicKey;
+    cert.serialNumber = String(Math.floor(Math.random() * 1024) + 1);
+
+    const date = new Date();
+    cert.validity.notBefore = date;
+    cert.validity.notAfter = new Date(date.getFullYear() + 1, date.getMonth(), date.getDay());
+
+    const attributes = [
+      {
+        name: 'organizationName',
+        value: 'XUD autogenerated certificate',
+      },
+      {
+        name: 'commonName',
+        value: hostname(),
+      },
+    ];
+
+    cert.setSubject(attributes);
+    cert.setIssuer(attributes);
+
+    cert.setExtensions([
+      {
+        name: 'subjectAltName',
+        altNames: [
+          {
+            type: 2,
+            value: 'localhost',
+          },
+          {
+            type: 7,
+            ip: '127.0.0.1',
```

Comment:
At some point I think we'll definitely want to support additional IP addresses/domain names in the cert, similar to how lnd has `tlsextradomain` and `tlsextraip`. Doesn't have to be in this PR tho, but maybe add a TODO and lets make sure to create an issue after this is merged.